src/mon/FSCommands.cc

```diff
@@ -268,6 +287,41 @@ class FsNewHandler : public FileSystemCommandHandler
       mon->osdmon()->wait_for_writeable(op, new PaxosService::C_RetryMessage(mon->mdsmon(), op));
       return -EAGAIN;
     }
+
+    bool recover = false;
+    cmd_getval(cmdmap, "recover", recover);
+
+    auto&& fs = fsmap.create_filesystem(fs_name, metadata, data,
+        mon->get_quorum_con_features(), fscid, recover);
+
+    // set fs options
+    int ret = 0;
+    string set_fsops_info;
+    int max_mds = 1;
+    for(int i = 1 ; i < (int)fsops_vec.size() ; i+=2) {
+      std::ostringstream oss;
+      ret = set_val(mon, fsmap, op, cmdmap, oss, fs_name, fsops_vec[i],
+                    fsops_vec[i+1]);
+      if(ret < 0) {
+        ss << oss.str();
+        break;
+      }
+      if((i + 2) <= (int)fsops_vec.size()) {
+        set_fsops_info.append("; ");
+      }
+      set_fsops_info.append(oss.str());
+      if(fsops_vec[i] == "max_mds") {
+        max_mds = stoi(fsops_vec[i+1]);
```

Comment:
Let's not interpret `max_mds` in two locations. Get the value directly from the `mds_map` in the code below instead.

Reply:
done